File: benchmarks/baselines/cross-language/matmul.go
package main

import "os"

func main() {
	n := 200
	a := make([]float64, n*n)
	b := make([]float64, n*n)
	for i := range a { a[i] = float64((i+1)%1000); b[i] = float64((i+1)%1000) }
	c := make([]float64, n*n)
	for r := 0; r < n; r++ {
		for col := 0; col < n; col++ {
			acc := 0.0
			for k := 0; k < n; k++ { acc += a[r*n+k] * b[k*n+col] }
			c[r*n+col] = acc
		}
	}
	if c[0] > 0 { os.Exit(0) } else { os.Exit(1) }
}

File: benchmarks/baselines/cross-language/matmul.py
import sys
n = 200
a = [float((i + 1) % 1000) for i in range(n * n)]
b = [float((i + 1) % 1000) for i in range(n * n)]
c = [0.0] * (n * n)
for r in range(n):
    for col in range(n):
        acc = 0.0
        for k in range(n):
            acc += a[r * n + k] * b[k * n + col]
        c[r * n + col] = acc
sys.exit(0 if c[0] > 0.0 else 1)
